src/components/global/PlaygroundTabs/index.tsx

```diff
@@ -0,0 +1,231 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React, { useState, cloneElement, isValidElement, type ReactElement, useRef, createRef, useEffect } from 'react';
+import useIsBrowser from '@docusaurus/useIsBrowser';
+import { useScrollPositionBlocker, duplicates } from '@docusaurus/theme-common';
+import type { Props } from '@theme/Tabs';
+import type { Props as TabItemProps } from '@theme/TabItem';
+
+import clsx from 'clsx';
+
+import styles from './styles.module.css';
+
+// A very rough duck type, but good enough to guard against mistakes while
+// allowing customization
+function isTabItem(comp: ReactElement): comp is ReactElement<TabItemProps & { icon?: JSX.Element }> {
+  return typeof comp.props.value !== 'undefined';
+}
+
+function TabsComponent(props: Props): JSX.Element {
```

Comment:
It seems like we are copy and pasting the Docusaurus Tabs implementation instead of extending it. Is there a way we can extend this without having to re-implement?

Reply:
Docusaurus does not expose access to manipulate the render functions that we are extending. We could use reflection, but I would advise against it. We could also roll our own tabs component (which is sorta what we are doing), but then we would need to go through the process of revalidating keyboard interactions and other design considerations that docusaurus has already validated in their codebase.

Reply:
Ok sounds good

Reply:
If we start using this pattern in more than just overlays (areas that currently I assume will need multi-file examples for the controller pattern); we could start to add spikes to own the implementation source more as well. AFAIK, most of this implementation could be a web component, since the content is slotted in.